tests/mixins/test_CheckNumericMixin.py

```diff
@@ -0,0 +1,25 @@
+import pytest
+
+from tests.test_data import create_df_2
+from tubular.mixins import CheckNumericMixin
+
+
+class TestCheckNumericMixin:
+    "tests for CheckNumericMixin class"
+
+    @pytest.mark.parametrize("library", ["pandas", "polars"])
+    def test_check_numeric_columns(self, library):
+        "test check_numeric_columns method raises appropriate error"
+
+        df = create_df_2(library=library)
```

Comment:
I think we should probably add in testing for more dtypes here.  Currently create_df_2 only has integers and strings but I think we should also cover floats, all None and np.nan at least?

Potential for pandas / polars to have different handling for which of these count as numeric so I think the narwhalification makes this more important than perhaps it was before.

Reply:
yeah fair point, added more cases here https://github.com/lvgig/tubular/pull/339/commits/f913f98bc365c93ca8a4951b22ddbf88ce316190

(this did spillover into a couple of other files)